unblob/finder.py

```diff
@@ -114,9 +120,9 @@ def _hyperscan_match(
     logger.debug("Found valid chunk", chunk=chunk, handler=handler.NAME, _verbosity=2)
     context.all_chunks.append(chunk)
 
-    # Terminate scan if we match till the end of the file
-    if chunk.end_offset == context.file_size:
-        logger.debug("Chunk covers till end of the file", chunk=chunk)
+    # Terminate scan if the whole file is matched
+    if chunk.start_offset == 0 and chunk.end_offset == context.file_size:
```

Comment:
This is also a heavy performance optiomization to match till the end of the file. It is a common pattern to have some prefix in a file followed by the content we pick up. Once we identify that the chunk covers till the end of the file, we can just bail out searching.